### PR TITLE
Added safeAfter to Maybe (#207)

### PR DESCRIPTION
**Pull request by karthikiyengar**
It's here!

I'm not sure if the examples are as good as they can be, so please feel free to extend/modify/suggest.

Also, I'd be happy to address any other comments.

### CODE REVIEW

**Comment by karthikiyengar** [comment]
Also, I'm not sure what went wrong with the super-long commit trail. Should I not be doing it this way?

File: docs/src/pages/docs/crocks/Maybe.md [hunk 1]
```diff
@@ -1053,6 +1053,64 @@ safe(isSmall, [ 1, 2 ])
...
+
+Often times, you might want to lift the result of a function into a `Maybe` based
+on some predicate. Similar to [`safe`](#safe), you pass either a `Pred` or a 
+predicate function, along with a unary function. `safeAfter` then gives you a new function, 
+which when invoked evaluates the predicate against the result of executing 
+the unary function. This result is then lifted into a `Just` if the predicate evaluates 
+to `true`, or a `Nothing` if it returns `false`
+
+```javascript
+import Pred from 'crocks/Pred'
```
Comment (evilsoft) [inline docs/src/pages/docs/crocks/Maybe.md]: This example does not match the example in the README and things in the docs structure are not linted (right now), so there is a chance the code may contain linting errors that go unnoticed. What I typically do is write the README first, and then copy it over to the docs folder when the README passes the linter.

File: src/Maybe/README.md [hunk 2]
```diff
@@ -1032,6 +1032,68 @@ safe(isSmall, [ 1, 2 ])
...
+
+Often times, you might want to lift the result of a function into a `Maybe` based
+on some predicate. Similar to [`safe`](#safe), you pass either a `Pred` or a 
+predicate function, along with a unary function. `safeAfter` then gives you a new function, 
+which when invoked evaluates the predicate against the result of executing 
+the unary function. This result is then lifted into a `Just` if the predicate evaluates 
+to `true`, or a `Nothing` if it returns `false`
+
+```javascript
+import Pred from 'crocks/Pred'
```
[the remark this answers is not in the document]
Comment (karthikiyengar) [reply src/Maybe/README.md]: This totally makes sense, my example was pretty pointless with regards to showing real-life use cases! I've used your example.

**Comment by evilsoft** [comment]
Yeah, I am a little 🌽cerned about those extra commits, looks like a bad rebase. and it rewrote history. mixing in your original commits on your forked master with commits from merging of PRs that in turn squashed those commits.
Typically on my fork I keep master in sync with upstream master and then do work on branches on my downstream, then submit those branches to the upstream as a pull request.

Looks like these PRs are all from commits on your master. So if we merged this now, before the fork is cleaned up, then it will bring in all the previous commits that we squashed when the PRs were merged.

**Comment by karthikiyengar** [comment]
Makes sense. I've opened another PR - https://github.com/evilsoft/crocks/pull/241

**Comment by evilsoft** [comment]
Closing in favor of https://github.com/evilsoft/crocks/pull/241/files

**Pull request closed without merging**